vars/neoDeploy.groovy

```diff
@@ -10,21 +11,73 @@ import static com.sap.piper.Prerequisites.checkScript
 
 @Field String STEP_NAME = getClass().getName()
 @Field Set GENERAL_CONFIG_KEYS = [
-    'neo'
+    'neo',
+    /**
+     * The Jenkins credentials containing user and password used for SAP CP deployment.
+     * @parentConfigKey neo
+     */
+    'credentialsId',
+    /**
+     * Name of the application you want to manage, configure, or deploy.
+     * @parentConfigKey neo
+     */
+    'application',
+    /**
+     * Name of SAP Cloud Platform application runtime.
+     * @parentConfigKey neo
+     */
+    'runtime',
+    /**
+     * Version of SAP Cloud Platform application runtime.
+     * @parentConfigKey neo
+     */
+    'runtimeVersion',
+    /**
+     * The SAP Cloud Platform host to deploy to.
+     * @parentConfigKey neo
+     */
+    'host',
+    /**
+     * The SAP Cloud Platform account to deploy to.
+     * @parentConfigKey neo
+     */
+    'account'
 ]
 @Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+    /** @see dockerExecute */
     'dockerEnvVars',
+    /** @see dockerExecute */
     'dockerImage',
+    /** @see dockerExecute */
     'dockerOptions',
+    /**
+     * The path to the `neo-java-web-sdk` tool used for SAP CP deployment.
+     * If no parameter is provided, the tool is expected on the `PATH`.
+     */
     'neoHome',
+    /** The path to the archive for deployment to SAP CP. If not provided `mtarFilePath` from commom pipeline environment is used instead.*/
     'source'
 ])
 
 @Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS.plus([
+    /** The deployment mode which should be used. Available options are:
+     * *`'mta'` - default,
+     * *`'warParams'` - deploying WAR file and passing all the deployment parameters via the function call,
+     * *`'warPropertiesFile'` - deploying WAR file and putting all the deployment parameters in a .properties file.
```

Comment:
There is a strange star `*` contained in the generated documentation. Only before `warPropertiesFile`, not for the other modes. Most likely a bug in the docu generator. We should fix that before we have a docu with a strange star ...

Depending on how urgent it gets to have the generated docu finished we can also merge without that fix, but in this case we should try to get that fixed in another PR soon ...

Reply:
This problem I did not have it, but is is not possible to have new lines, maybe we could implement it in another PR.